exporters/otlp/otlptrace/internal/otlpconfig/options.go

```diff
@@ -244,11 +248,37 @@ func NewGRPCOption(fn func(cfg Config) Config) GRPCOption {
 
 func WithEndpoint(endpoint string) GenericOption {
 	return newGenericOption(func(cfg Config) Config {
-		cfg.Traces.Endpoint = endpoint
+		// Add scheme if not present
+		if !hasScheme(endpoint) {
+			endpoint = (&url.URL{Scheme: getScheme(cfg), Host: endpoint}).String()
+		}
+		u, err := url.Parse(endpoint)
```

Comment:
If no scheme is provided, we generate a new URL that we move to a string, and then we parse to a URL again.
Could we avoid doing that round twice by creating a `var u *url.URL` before the condition, and then setting it to a custom value, or parsing from the string?